bootstrap.py

```diff
@@ -1434,9 +1436,15 @@ def exec_service(service, command, failonerror=True):
 
         # Configure new proxy_id for Puppet (if not skipped), and OpenSCAP (if available and not skipped)
         if 'foreman' not in options.skip and 'puppet' not in options.skip:
-            print_running("Calling Foreman API to update Puppet master and Puppet CA for %s to %s" % (FQDN, options.foreman_fqdn))
-            update_host_capsule_mapping("puppet_proxy_id", smart_proxy_id, current_host_id)
-            update_host_capsule_mapping("puppet_ca_proxy_id", smart_proxy_id, current_host_id)
+            if puppet_master_smart_proxy_id is not None and puppet_ca_smart_proxy_id is not None:
+                print_running("Calling Foreman API to update Puppet Master for %s to %s" % (FQDN, options.puppet_server))
+                update_host_capsule_mapping("puppet_proxy_id", puppet_master_smart_proxy_id, current_host_id)
+                print_running("Calling Foreman API to update Puppet CA for %s to %s" % (FQDN, options.puppet_ca_server))
+                update_host_capsule_mapping("puppet_ca_proxy_id", puppet_ca_smart_proxy_id, current_host_id)
+            else:
```

Comment:
I wonder if we need this branching here. The `if/else` is only here to generate slightly different output, right? I think the user won't lose much if we drop the `to %s` part of the messages and then we can drop the whole `if/else` completely if we set `puppet_master_smart_proxy_id` and `puppet_ca_smart_proxy_id` to `smart_proxy_id` if the search didn't return anything.

Now, after writing this, I wonder if we need to set these to the Smart Proxy at all if the user asked for an external (not-found) Puppet setup. Because if the user bootstraps a machine against `foreman.example.com`, but is using `puppet.example.com` for Puppet Master and CA, why would we set the Puppet options in Foreman to `foreman.example.com`? That'd be wrong, no?

@sideangleside thoughts?